clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -20,26 +20,35 @@
  * This is used to describe per-partition state in the MetadataResponse.
  */
 public class PartitionInfo {
+    public static final int UNKNOWN_LEADER_EPOCH = -1;
     private final String topic;
     private final int partition;
     private final Node leader;
+    private final int leaderEpoch;
```

Comment:
I think it's more aligned with the existing Kafka interface to use `Optional<Integer> leaderEpoch`. For example, `org.apache.kafka.clients.consumer.OffsetAndMetadata`.